dev.goats.xivlauncher.yml

```diff
@@ -1,31 +1,38 @@
 app-id: dev.goats.xivlauncher
+
 runtime: org.freedesktop.Platform
 runtime-version: '22.08'
 sdk: org.freedesktop.Sdk
 sdk-extensions:
   - org.freedesktop.Sdk.Extension.dotnet6
+
 rename-icon: xivlauncher
 copy-icon: true
+
 build-options:
   append-path: "/usr/lib/sdk/dotnet6/bin"
   append-ld-library-path: "/usr/lib/sdk/dotnet6/lib"
   append-pkg-config-path: ":/usr/lib/sdk/dotnet6/lib/pkgconfig"
   env:
     DOTNET_CLI_TELEMETRY_OPTOUT: "true"
 command: xivlauncher
+
 finish-args:
   - --share=ipc
-  - --socket=fallback-x11
-  #- --socket=wayland re-add later when libdecor support is added to builds
+  - --socket=x11 # Wine doesn't support Wayland yet.
+  - --socket=wayland # XIVLauncher does, via ImGui's SDL2 backend.
   - --share=network
-  - --filesystem=home
+  - --filesystem=~/.xlcore # Ideally, XIVLauncher would use the appropriate xdg-specs rather than hardcoding this path.
```

Comment:
We need to allow access to more than just `~/.xlcore` otherwise the launcher won't correctly function when needing to authenticate with steam, and wine also needs to create symlinks outside of `~/.xlcore` as well.

Need to allow access to:

- xdg-pictures:rw
- xdg-documents:rw
- xdg-videos:rw
- xdg-music:rw
- xdg-desktop:rw
- xdg-downloads:rw:
- home/.steam:rw
- xdg-data/Steam:rw

Ideally, we wouldn't be symlinking wineprefix directories outside of the prefix, but if we continue doing this we'll have to allow access to these directories otherwise everything will just be placed in weird and inconsistent places (or just run into errors)

Reply:
Wine has an option to sandbox the home directory iirc, so it won't symlink those; additionally, I'm going to be changing the permissions to use --persist so that it doesn't put stuff directly in the home folder (which is discouraged by Flathub, as flatpaks should be self contained)